packages/mongo/collection.js

```diff
@@ -461,149 +574,54 @@ var throwIfSelectorIsNotId = function (selector, methodName) {
     }
+    throw e;
+  }
+};
 
-    // XXX see #MeteorServerNull
-    if (self._connection && self._connection !== Meteor.server) {
-      // just remote to another endpoint, propagate return value or
-      // exception.
-
-      var enclosing = DDP._CurrentInvocation.get();
-      var alreadyInSimulation = enclosing && enclosing.isSimulation;
-
-      if (Meteor.isClient && !wrappedCallback && ! alreadyInSimulation) {
-        // Client can't block, so it can't report errors by exception,
-        // only by callback. If they forget the callback, give them a
-        // default one that logs the error, so they aren't totally
-        // baffled if their writes don't work because their database is
-        // down.
-        // Don't give a default callback in simulation, because inside stubs we
-        // want to return the results from the local collection immediately and
-        // not force a callback.
-        wrappedCallback = function (err) {
-          if (err)
-            Meteor._debug(name + " failed: " + (err.reason || err.stack));
-        };
-      }
-
-      if (!alreadyInSimulation && name !== "insert") {
-        // If we're about to actually send an RPC, we should throw an error if
-        // this is a non-ID selector, because the mutation methods only allow
-        // single-ID selectors. (If we don't throw here, we'll see flicker.)
-        throwIfSelectorIsNotId(args[0], name);
-      }
-
-      ret = chooseReturnValueFromCollectionResult(
-        self._connection.apply(self._prefix + name, args, {returnStubValue: true}, wrappedCallback)
-      );
-
-    } else {
-      // it's my collection.  descend into the collection object
-      // and propagate any exception.
-      args.push(wrappedCallback);
-      try {
-        // If the user provided a callback and the collection implements this
-        // operation asynchronously, then queryRet will be undefined, and the
-        // result will be returned through the callback instead.
-        var queryRet = self._collection[name].apply(self._collection, args);
-        ret = chooseReturnValueFromCollectionResult(queryRet);
-      } catch (e) {
-        if (callback) {
-          callback(e);
-          return null;
-        }
-        throw e;
-      }
-    }
+// Determine if this collection is simply a minimongo representation of a real
+// database on another server
+Mongo.Collection.prototype._isRemoteCollection = function _isRemoteCollection() {
+  // XXX see #MeteorServerNull
+  return this._connection && this._connection !== Meteor.server;
+}
+
+// Convert the callback to not return a result if there is an error
+function wrapCallback(callback, convertResult) {
+  if (!callback) {
+    return;
+  }
 
-    // both sync and async, unless we threw an exception, return ret
-    // (new document ID for insert, num affected for update/remove, object with
-    // numberAffected and maybe insertedId for upsert).
-    return ret;
-  };
-});
+  if (convertResult) {
```

Comment:
It would be a tad shorter to do `convertResults = convertResults || _.identity` (or even can you do that as a default argument value in ES5?)

Anyhow, not important enough to change now, it just came to mind.